sample/Evernote Business/server.py

```diff
@@ -0,0 +1,233 @@
+
+
+@app.route("/", methods=['POST','GET'])
+def main():
+	"""Takes the access token in the session and presents the user with notes that are tagged "template" or "Template" 
+
+	If there are no notes tagged "template" or "Template" 4 example notes are created
+
+	If the user is not logged in: a splash page with the option to authorize the appliction is presented
+	"""
+
+	#check to see the user has logged in
+	if "access_token" in session.keys():
+		
+		try:
+			#setup Evernote client
+			client = EvernoteClient(token=session["access_token"], sandbox=sandbox)
+			user_store = client.get_user_store()
+			note_store = client.get_note_store()			
+		except Errors.EDAMSystemException:
+			#if the user authentication token fails to work prompt them to reautenticate
+			return render_template("error.html", error_message="Your session is no longer valid.  Please click here <a href=\'/clear'>click here</a> to reset and login again.")
+
+		#if the user is not a part of the business, tell the user
+		user = user_store.getUser()
+		if not user.accounting.businessId:
+			return render_template("error.html", error_message="This account is not a part of a business.  Please click here <a href=\'/clear'>click here</a> to reset and try another account.")
+		else:
+			business_store = client.get_business_note_store()
+
+		#setup search
+		notebook_filter=NoteStoreTypes.NoteFilter()
+		result_spec = NoteStoreTypes.NotesMetadataResultSpec(includeTitle=True)
+
+		# Start seraching for notes in the last day
+		# if less than 5 results increase to 10, 100, etc. until more than 5 are found or 
+		# the number results between iterations are the same
+		search_results_len = None
+		while search_results_len<10:
+			#setup counter
+			try:
+				counter += 1
+			except NameError:
+				counter = 0
+			#define number of pervious days to search:
+			days_pervious_to_search = str(int(math.pow(10, counter)))
+			#Define search grammer (docs: https://dev.evernote.com/doc/articles/search_grammar.php)
+			notebook_filter.words = "created:day-"+days_pervious_to_search+" updated:day-"+days_pervious_to_search
+
+			#perform search
+			search_results = business_store.findNotesMetadata(notebook_filter,0 , 40000, result_spec)
+			
+			#break if you get the same number of results for a order of magnitude increase in the number of days
+			# (assume that is the total number of search results present)
+			if len(search_results.notes)==search_results_len and search_results_len != 0:
+				break
+
+			# if you search the last 10,000 days and there are not business notes
+			# assume there are no business notes and inform the user
+			if counter == 4 and search_results_len==0:
+				return render_template("error.html", error_message="This account has no business notes!  Please click please add and sync some business notes and refresh the page.")
+
+			search_results_len = len(search_results.notes)
+
+		notes_metadata = search_results.notes
+
+		auth_result = user_store.authenticateToBusiness()
```

Comment:
The `business_store` already authenticated to the Business in order to call `findNotesMetadata()` - is there a way to extract that auth token (and the related parts of the authentication result) from the `business_store` object instead of calling `authenticateToBusiness()` again?